src/util/arr.h

```diff
@@ -111,7 +111,6 @@ static inline uint32_t array_len(array_t arr) {
 
 static inline void *array_trimm(array_t arr, uint32_t len, uint32_t cap) {
   array_hdr_t *arr_hdr = array_hdr(arr);
-  assert(len >= 0 && "trimming len is negative");
```

Comment:
Why?

Reply:
`len` is an unsigned int, so this can never trigger.

Reply:
👍